### ISSUE TITLE
FAILED HostConnectionPoolSpec.NewPoolImplementation on AkkaHttpEngineNoNetwork should create a new connection when previous one was closed regularly between requests

### ISSUE THREAD

**Issue by jlprat** [akka/akka-http#1751]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/2078/testReport/junit/akka.http.impl.engine.client/HostConnectionPoolSpec/NewPoolImplementation_on_AkkaHttpEngineNoNetwork_should_create_a_new_connection_when_previous_one_was_closed_regularly_between_requests/

Full log for it: https://jenkins.akka.io:8498/job/pr-validator-akka-http/2078/consoleFull

Found in #1750
 
```
Error Message

Expected OnNext(_), yet no element signaled during 4500 milliseconds

Stacktrace

java.lang.AssertionError: Expected OnNext(_), yet no element signaled during 4500 milliseconds
	at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:325)
	at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:316)
	at akka.stream.testkit.TestSubscriber$Probe.requestNext(StreamTestKit.scala:753)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$TestSetup$1.expectResponse(HostConnectionPoolSpec.scala:380)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$TestSetup$1.expectResponseEntityAsString(HostConnectionPoolSpec.scala:385)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$$anonfun$apply$mcV$sp$15$$anon$11.<init>(HostConnectionPoolSpec.scala:261)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$$anonfun$apply$mcV$sp$15.apply(HostConnectionPoolSpec.scala:247)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$$anonfun$apply$mcV$sp$15.apply(HostConnectionPoolSpec.scala:247)
```

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/2088/testReport/junit/akka.http.impl.engine.client/HostConnectionPoolSpec/NewPoolImplementation_on_AkkaHttpEngineTCP_should_create_a_new_connection_when_previous_one_was_closed_regularly_between_requests/

**Comment by jrudolph** [comment]
Probably fixed in #1776.
